src/components/MyAccount/SettingsTab/PasswordChangeForm.tsx

```diff
@@ -0,0 +1,110 @@
+import { useState } from "react"
+import {
+  Form,
+  FormField,
+  TextInput,
+  Button,
+} from "@nypl/design-system-react-components"
+import styles from "../../../../styles/components/MyAccount.module.scss"
+import { BASE_URL } from "../../../config/constants"
+import type { Patron } from "../../../types/myAccountTypes"
+
+const PasswordChangeForm = ({
+  patron,
+  updateModal,
+}: {
+  patron: Patron
+  updateModal: (errorMessage?: string) => void
+}) => {
+  const [formData, setFormData] = useState({
+    oldPassword: "",
+    newPassword: "",
+    confirmPassword: "",
+    passwordsMatch: true,
+  })
+
+  const validateForm = () => {
+    return (
+      formData.oldPassword !== "" &&
+      formData.newPassword !== "" &&
+      formData.confirmPassword !== "" &&
+      formData.passwordsMatch
+    )
+  }
+
+  const handleInputChange = (e) => {
+    const { id, value } = e.target
+    if (id === "confirmPassword") {
+      formData.passwordsMatch = formData.newPassword === value
```

Comment:
State should not be updated directly like this. Include this as part of the `setFormData` function on line 40.